honeybee_energy/schedule/fixedinterval.py

```diff
@@ -0,0 +1,690 @@
+
+    @name.setter
+    def name(self, name):
+        self._name = valid_ep_string(name, 'schedule fixed interval name')
+
+    @property
+    def values(self):
+        """Get or set the schedule's numerical values, which occur at a fixed interval.
+        """
+        return self._values
+
+    @values.setter
+    def values(self, values):
+        self._values = self._check_values(values)
+
+    @property
+    def schedule_type_limit(self):
+        """Get or set a ScheduleTypeLimit object used to assign units to schedule values.
+        """
+        return self._schedule_type_limit
+
+    @schedule_type_limit.setter
+    def schedule_type_limit(self, schedule_type):
+        if schedule_type is not None:
+            assert isinstance(schedule_type, ScheduleTypeLimit), 'Expected ' \
+                'ScheduleTypeLimit for ScheduleRuleset schedule_type_limit. ' \
+                'Got {}.'.format(type(schedule_type))
+        self._schedule_type_limit = schedule_type
+
+    @property
+    def out_of_range_value(self):
+        """Get or set the value to be used for all times not covered by the input values.
+        """
+        return self._out_of_range_value
+
+    @out_of_range_value.setter
+    def out_of_range_value(self, value):
+        self._out_of_range_value = float_in_range(
+            value, input_name='schedule fixed interval out_of_range_value')
+
+    @property
+    def interpolate(self):
+        """Get or set a boolean noting whether values should be interpolated."""
+        return self._interpolate
+
+    @interpolate.setter
+    def interpolate(self, interpolate):
+        self._interpolate = bool(interpolate)
+
+    @property
+    def timestep(self):
+        """Get the integer for the schedule's number of steps per hour."""
+        return self._timestep
+
+    @property
+    def start_date(self):
+        """Get the ladybug Date object noting when the the input values take effect."""
+        return self._start_date
+
+    @property
+    def end_date_time(self):
+        """Get a ladybug DateTime object for the end time of the schedule's values."""
+        num_hoys = (len(self._values) - 1) / self.timestep
+        end_hoy = (self.start_date.doy - 1) * 24 + num_hoys
+        if not self.is_leap_year:
+            end_dt = DateTime.from_hoy(end_hoy) if end_hoy < 8760 else \
+                DateTime.from_hoy(end_hoy - 8760)
+        else:
+            end_dt = DateTime.from_hoy(end_hoy, True) if end_hoy < 8784 else \
+                DateTime.from_hoy(end_hoy - 8760, True)
+        return end_dt
+
+    @property
+    def is_leap_year(self):
+        """Get a boolean noting whether the schedule is over a leap year.
+
+        Note that this property originates from the leap_year property on the
+        input start_date.
+        """
+        return self._start_date.leap_year
+
+    @property
+    def data_collection(self):
+        """A DataCollection of schedule values at the native start_date and timestep."""
```

Comment:
what is a _native_ start date?

Reply:
Its the `start_date` assigned to the schedule (as opposed to a user-input `start_date`. I can make this clearer in the docstring.